src/components/vm/disks/vmDiskColumns.jsx

```diff
@@ -105,3 +115,81 @@ DiskExtras.propTypes = {
     errorPolicy: PropTypes.string,
     idPrefix: PropTypes.string.isRequired,
 };
+
+export const DiskActions = ({ vm, vms, disk, supportedDiskBusTypes, idPrefixRow }) => {
+    const [isActionOpen, setIsActionOpen] = useState(false);
+
+    const Dialogs = useDialogs();
+
+    const onRemoveDisk = () => {
+        return domainDetachDisk({ connectionName: vm.connectionName, id: vm.id, name: vm.name, target: disk.target, live: vm.state === 'running', persistent: vm.persistent })
+                .then(() => domainGet({ connectionName: vm.connectionName, id:vm.id }));
+    };
+
+    const deleteDialogProps = {
+        title: _("Remove disk from VM?"),
+        actionName: _("Remove"),
+        errorMessage: cockpit.format(_("Disk $0 could not be removed"), disk.target),
+        actionDescription: cockpit.format(_("This disk will be removed from $0:"), vm.name),
+        objectDescription: [
+            { name: _("Target"), value: <span className="ct-monospace">{disk.target}</span> },
+            ...DISK_SOURCE_LIST.flatMap(entry => getDiskSourceValue(disk.source, entry.name)
+                ? { name: entry.label, value: <span className="ct-monospace">{getDiskSourceValue(disk.source, entry.name)}</span> }
+                : [])
+        ],
+        deleteHandler: () => onRemoveDisk(),
```

Comment:
You can just write `deleteHandler: onRemoveDisk`

Reply:
Fixed